src/RustyObjectStore.jl

```diff
@@ -81,52 +98,203 @@ function init_object_store(config::ObjectStoreConfig=DEFAULT_CONFIG)
     return nothing
 end
 
+"""
+    $TYPEDEF
+
+# Keyword Arguments
+- `request_timeout_secs::Option{Int}`: (Optional) Client request timeout in seconds.
+- `connect_timeout_secs::Option{Int}`: (Optional) Client connection timeout in seconds.
+- `max_retries::Option{Int}`: (Optional) Maximum number of retry attempts.
+- `retry_timeout_secs::Option{Int}`: (Optional) Maximum amount of time from the initial request after which no further retries will be attempted (in seconds).
+"""
+struct ClientOptions
+    params::Dict{String, String}
+
+    function ClientOptions(;
+        request_timeout_secs::Option{Int} = nothing,
+        connect_timeout_secs::Option{Int} = nothing,
+        max_retries::Option{Int} = nothing,
+        retry_timeout_secs::Option{Int} = nothing
+    )
+        params = Dict()
+        if !isnothing(request_timeout_secs)
+            params["timeout"] = "$(string(request_timeout_secs))s"
+        end
+
+        if !isnothing(connect_timeout_secs)
+            params["connect_timeout"] = "$(string(connect_timeout_secs))s"
```

Comment:
why do `connect_timeout_secs` and `request_timeout_secs` need a trialing `s`, but `retry_timeout_secs` doesn't?

Reply:
`request_timeout_secs` and `connect_timeout_secs` are parsed by the object_store library directly into duration objects and it expects something like `30s` meaning 30 seconds.

The `retry_timeout_secs` is parsed by us on the Rust side (because we are making up for a parameter that is not exposed in `object_store`'s API) and we assume the input will always be just the number of seconds.

It would be good to use the same parsing as the other ones but the parsing function is private and I didn't think it warranted an implementation (maybe a crate already exists, I didn't check).

Do you think we should do anything differently here?

Reply:
My vote would be to not complicate the parsing on the rust side if we don't have to.

Reply:
Okay, makes sense... i think i'd add a comment saying why we do this

maybe on the two with the trailing `s`, something like
```
# Include `s` so Rust `Duration` knows to parse this as seconds
```
then on the other one
```
# Parsed directly by `object_store_ffi` as seconds, so don't need to include trailing `s`
```
or something along those lines 

otherwise it seems a bit surprising and inconsistent